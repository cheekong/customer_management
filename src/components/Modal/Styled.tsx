import styled from 'styled-components';

export const StyledModal = styled.div`
    background-color:white;
    margin-top: 10%;
    max-width: 400px;
    height:300px;
    width: 40%;
    margin-left: auto;
    margin-right: auto;
    border-radius: 4px;
`;

export const StyledModalBody = styled.div`
    width: 60%;
    height: 200px;
    line-height: 20px;
    margin-top:50px;
    margin-left: auto;
    margin-right: auto;
`;