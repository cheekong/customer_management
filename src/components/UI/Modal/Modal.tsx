import React from 'react';
import styled from 'styled-components'

const Modal: React.SFC  = () => {
    return (
      <div className="App">
         modal
      </div>
    );
  }
  
  export default Modal;
  