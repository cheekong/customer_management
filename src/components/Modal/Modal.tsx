import React, { ReactNode } from 'react';

import Backdrop from '../UI/Backdrop/Backdrop'
import Form from '../UI/Form/Form';
import List from '../UI/List/List';
import ListItem from '../UI/List/ListItem/ListItem';
import {StyledModal, StyledModalBody} from './Styled';

interface MyProps {
  callbackOnClose?: any;
  show: boolean;
  title: string;
  children: ReactNode
}

interface MyState {
  show: false;
}

class Modal extends React.Component<MyProps, MyState> {

  componentWillUnmount() {
    if(this.props.callbackOnClose){
      this.props.callbackOnClose();
    }
  }

    render(){
        let content = null; 
        
        if(this.props.show){
            content = (
                <Backdrop>
                    <StyledModal>
                      <Form title={this.props.title}>
                          <StyledModalBody>
                            {this.props.children}
                          </StyledModalBody>
                      </Form>
                    </StyledModal>
                </Backdrop>
            )
        }
        
        return content
    }
}

export default Modal;