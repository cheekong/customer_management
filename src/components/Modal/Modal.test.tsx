import React from 'react';
import Modal from './Modal';
import renderer from 'react-test-renderer';

describe('Modal', () => {
    test('it works', () => {
    
        const component = renderer.create(
            <Modal 
                show
                title='my modal'
            >
                <p>TEST</p>
            </Modal>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})


