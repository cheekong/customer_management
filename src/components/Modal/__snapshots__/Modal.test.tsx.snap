// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal it works 1`] = `
<div
  className="sc-bdVaJa fLIace"
>
  <div
    className="sc-ifAKCX iiGFiZ"
  >
    <form
      className="sc-bwzfXH inPTmq"
      onSubmit={[Function]}
    >
      <div
        className="sc-htpNat jqIXVP"
      >
        <h1>
          my modal
        </h1>
      </div>
      <div
        className="sc-bxivhb kinwFJ"
      >
        <div
          className="sc-EHOje efBfVR"
        >
          <p>
            TEST
          </p>
        </div>
      </div>
    </form>
  </div>
</div>
`;
